apps/charterafrica/src/components/Datasets/Datasets.js

```diff
@@ -0,0 +1,119 @@
+import { Section } from "@commons-ui/core";
+import { Box, LinearProgress } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { useState, useEffect, useRef, useImperativeHandle } from "react";
+
+import DatasetCard from "./DatasetCard";
+import DatasetFilterBar from "./DatasetFilterBar";
+import useDatasets from "./useDatasets";
+
+import NextPrevPagination from "@/charterafrica/components/NextPrevPagination";
+import queryString from "@/charterafrica/utils/datasets/queryString";
+
+const Datasets = React.forwardRef(function Datasets(
+  {
+    sx,
+    data: originalDatasets,
+    labels,
+    tags = [],
+    countries = [],
+    totalPages: originalTotalPages,
+    sortOptions = [],
+  },
+  ref
+) {
+  const [page, setPage] = useState(1);
+  const [q, setQ] = useState("");
+  const [sort, setSort] = useState("");
+  const [selectedCountries, setSelectedCountries] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [totalPages, setTotalPages] = useState(originalTotalPages);
+  const [datasets, setDatasets] = useState(originalDatasets || []);
+  const router = useRouter();
+  const datasetsRef = useRef();
+  useImperativeHandle(ref, () => datasetsRef.current);
+  const { asPath } = router;
+
+  const handleChangePage = (_, value) => setPage(value);
+  const handleChangeQ = (_, value) => setQ(value);
+  const handleChangeSort = (_, value) => setSort(value);
+  const handleChangeCountries = (_, value) => {
+    setSelectedCountries(value);
+  };
+
+  const handleChangeTags = (_, value) => setSelectedTags(value);
+
+  const query = queryString({
+    page,
+    q,
+    sort,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  useEffect(() => {
+    const pathname = asPath.split("?")[0];
+    router.push({ pathname, query }, undefined, {
+      scroll: false,
+      shallow: true,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query]);
+
+  const { data, isLoading } = useDatasets({
+    q,
+    sort,
+    page,
+    countries: selectedCountries,
+    tags: selectedTags,
+  });
+
+  if (isLoading && datasetsRef.current) {
+    datasetsRef.current.scrollIntoView({ behavior: "smooth" });
+  }
+
+  useEffect(() => {
+    if (!isLoading) {
+      const { datasets: filteredDatasets, totalPages: filteredTotalPages } =
+        data || {};
+      setDatasets(filteredDatasets);
+      setTotalPages(filteredTotalPages);
+    }
+  }, [data, isLoading]);
+
+  return (
+    <Box bgcolor="common.white" sx={sx} ref={datasetsRef}>
```

Comment:
Make sure this component has `scroll-margin` so that when next/prev is clicked, the full `FilterBar` is visible. See `Articles` component.